### PR TITLE
Fix analysis issues with Flutter v3.22

### PR DESCRIPTION
**Pull request by Amir-P**

### CODE REVIEW

**Review by amantoux — CHANGES_REQUESTED** [review]
Thanks @Amir-P 
We also need to update the CI

**Comment by Amir-P** [comment]
> Thanks @Amir-P 
> 
> We also need to update the CI

Thanks! Completely forgot about that. Now we have failing tests.

File: packages/fleather/lib/src/rendering/editor.dart [hunk 1]
```diff
@@ -885,7 +885,10 @@ class RenderEditor extends RenderEditableContainerBox
         final caretOffset = child.getOffsetForCaret(localPosition);
         final testPosition = TextPosition(offset: sibling.node.length - 1);
         final testOffset = sibling.getOffsetForCaret(testPosition);
-        final finalOffset = Offset(caretOffset.dx, testOffset.dy);
+        // The addition of 1 point to testOffset.dy is added because somehow in
+        // Flutter 3.22 the text painer for the offset is yielding wrong
```
Comment (amantoux) [inline packages/fleather/lib/src/rendering/editor.dart:889]: typo -> painter

File: packages/fleather/lib/src/rendering/editor.dart [hunk 2]
```diff
@@ -885,7 +885,10 @@ class RenderEditor extends RenderEditableContainerBox
...
+        // text position (1 line above the correct line).
+        final finalOffset = Offset(caretOffset.dx, testOffset.dy + 1);
```
Comment (amantoux) [inline packages/fleather/lib/src/rendering/editor.dart:891]: Really strange, no explanation for this?

**Review by amantoux — COMMENTED** [review]
Thanks @Amit-P. LGTM
Just one question (and a typo)

File: packages/fleather/lib/src/rendering/editor.dart (hunk 2 above)
Comment (Amir-P) [reply packages/fleather/lib/src/rendering/editor.dart:891]: Couldn't find any. I even took a quick look at the commits in framework and engine.

**Review by amantoux — APPROVED** [review]
LGTM

**Pull request merged**